src/modifiers/nullable.ts

```diff
@@ -2,6 +2,12 @@ import { refine } from '../refine';
 import { typeToPrint } from '../stringify';
 
 export const nullable = refine(
-  (value, t) => (value === null ? t.right(null) : t.nextNotValid(value)),
+  (value, t) =>
+    value === null
+      ? t.right(null)
+      : t.nextNotValid({
+          expected: 'nullable',
```

Comment:
In case of nullable string we should somehow return the information about string as well

Reply:
I don't know what you mean. Can you elaborate?

Reply:
I've added a few tests to explain what I meant:

https://github.com/typeofweb/schema/blob/65d9e173da4815c1f25751cac0b3b9712be170ec/__tests__/errors.spec.ts#L236-L305

## Examples 
Let's take a closer look at a few examples:

### Nullable, optional – alternatives

```ts
expectToMatchError(() => λ(string, nullable, validate)(123), {
  expected: 'string | null',
  got: 123,
});
```

In this case we expect a nullable string so the error should mention that. Some of the possible solutions are:
- `'string | null'`
- `nullable(string)`
- `['nullable', 'string']`
- … other ?

Same goes for `optional` and `nil`.

### Additional validation
`even` is a custom refinement that adds additional validation after the `number` validator. It's not an alternative like nullable or optional above.

```ts
expectToMatchError(() => λ(even, number, validate)(1), { expected: 'number(even)', got: 1 });
```

Some possible solutions are:
- `'number(even)'`
- `['number', 'even']`

**Question**: How do we distinguish between these and `nullable(string)` or `['nullable', 'string']`?

### Parsing before validating
`allowTimestamps` is a custom refinement which converts timestamps into Dates **before** the `date` validator runs.

```ts
expectToMatchError(() => λ(date, allowTimestamps, validate)(''), {
  expected: 'allowTimestamps(date)',
  got: '',
});
```

Some possible solutions are:
- `'allowTimestamps(date)'`
- `['allowTimestamps', 'date']`

Reply:
The ultimate goal of my work here is to be able to map errors from this library to user understandable errors via some function like this one:
```tsx
function errorToMessage(error: ErrorData): string {
  switch (error.expected) {
    case 'minStringLength':
      return `Enter at least ${error.args[0]} characters`;
    case 'max':
      return `Maximum allowed value is ${error.args[0]}`;
    case 'min':
      return `Minimum allowed value is ${error.args[0]}`;
    default:
      return 'This field is required.';
  }
}
```
making expected to something like `'string | null'` makes it much harder.

For me, the error detail should be bound to exact not fulfilled validator. The overall error indeed can log 'string | null" but the details should point to the exact invalid validator. In the case of `pipe(string, minLengthString(20))`:
* value: `undefined` - the `string` validation is failing because not string is provided
* value `'something'` - the `minLengthString` is failing because the provided input is too short.

WDYT?

Reply:
You're right!